database.go

```diff
@@ -59,6 +59,25 @@ func isKnownArch(arch string) bool {
 	return false
 }
 
+// normalizeOSAndVersion will only include the OSVersion if present.
+// The formart of OS on the platform specifier is <OS>[(<OSVersion>)]
+// OSVersion is optional only and is currently used only by windows OS.
+// If OsVersion is not specified, empty string is returned.
```

Comment:
(Two different ideas about this function description)
"say what the function does" version.
```suggestion
// normalizeOSAndVersion splits the provided platform specifier OS segment into an OS
// and OSVersion. The expected format is `<os>[(<OSVersion>)]`, e.g., `windows(10.0.17763)`.
// If `<os>` is not provided, the current host OS is returned as the first value.
// If optional `(<OSVersion>)` is not provided, an empty string is returned as the second value.
```

I'd also suggest naming the return values so the function signature is easier to read, since both return values have the same type.